File: Cpp/Unit01/Exercise08/src/exercise08.cpp
#include<iostream>
#include<string>
#include<fstream>

using namespace std;

int main(int argc, char** argv) {
    // Getting the path to the Input/Output folders
    string inputPath;
    string outputPath;
    if (argv[1] == NULL) {
        inputPath = "Inputs/input.txt";
        outputPath = "Outputs/output.txt";
    } else {
        string fs(argv[1]);
        inputPath = fs + "Inputs/input.txt";
        outputPath = fs + "Outputs/output.txt";
    }

    // Opening input and output files
    ifstream infile;
    ofstream outfile;
    infile.open(inputPath);
    outfile.open(outputPath);

    string str;
    float n;
    // Looping over the values, squaring them, and writing them out.
    while ( getline (infile,str) )
    {
      n = std::stof(str);
      outfile << n*n << "\n";
    }

    // Closing the files
    infile.close();
    outfile.close();

    return 0;
}

File: Fortran90/Unit01/Exercise08/src/exercise08.f90
program exercise08
    implicit none

    character(len=260) :: path, inputPath, outputPath
    real :: a
    integer :: io

    ! Creating the paths
    call get_command_argument(1, path)
    inputPath = trim(path)//"Inputs/input.txt"
    outputPath = trim(path)//"Outputs/output.txt"

    ! Opening the input and output files
    open(1, file=inputPath)
    open(2, file=outputPath)

    ! Reading in the input data, squaring, and writing
    !   to output file
    do 
        read(1,*,iostat=io) a
        if (io/=0) exit
        write(2,*) a*a
    end do

    ! Closing the files
    close(1)
    close(2)

end program exercise08